azext_edge/edge/providers/check/base.py

```diff
@@ -735,21 +735,20 @@ def add_display_and_eval(
 def get_resource_metadata_property(resource: Union[dict, Any], prop_name: str) -> Union[str, None]:
     if isinstance(resource, dict):
         return resource.get("metadata", {}).get(prop_name)
-    return getattr(resource.metadata, prop_name, None)
+    return getattr(resource.metadata, prop_name, None) if hasattr(resource, "metadata") else None
 
 
-def resources_grouped_by_namespace(resources: List[dict]):
-    resources.sort(key=lambda resource: get_resource_metadata_property(resource, prop_name="namespace"))
-    return groupby(resources, lambda resource: get_resource_metadata_property(resource, prop_name="namespace"))
+def key_func(resource):
+    return get_resource_metadata_property(resource, prop_name="namespace")
```

Comment:
My previous comment was assuming we'd define the `key_func` (just a suggested name) inside the `x_grouped_by_y` function:

```
def resources_grouped_by_namespace(resources):
    def by_namespace(item):
        return item.namespace
    resources.sort(key=by_namespace)
    ...
```

But if there are multiple functions that use this, we should keep it at the top level but rename it something a little more readable like "get_namespace"

Reply:
got it! I moved it upper level since the function defined inside is a bit repetitive, will update the name!